lib/steps.js

```diff
@@ -455,6 +457,57 @@ function unzip(passon) {
     });
 }
 
+function getTextFiles(passon) {
+    return new Promise((fulfill, reject) => {
+        debug('[%s] Checking file encoding', passon.id);
+        //regular expression for text file types (.txt .r .rmd .text .json .yml .yaml ..)
+        let textFileRegex = new RegExp(config.encoding.textFileRegex, 'i');
+        passon.textFiles = [];
+
+        //ignore text files
+        function ignoreFunc(file, stats) {
+          return !stats.isDirectory() && !textFileRegex.test(path.extname(file));
+        }
+
+        //recursively iterate through bagpath and collect file names
+        recursiveReaddir(passon.bagpath, [ignoreFunc], function (err, files) {
+            if (err) {
+                debug('[%s] Error reading text files: %s', passon.id, err.message);
+                reject(err);
+            }
+            debug('[%s] Found %s text files.', passon.id, files.length);
+            passon.textFiles = files;
+            fulfill(passon);
+        });
+    });
+}
+
+function checkEncoding(passon) {
+    return new Promise((fulfill, reject) => {
+        debug('[%s] Checking file encoding', passon.id);
+
+        for (var i = 0; i < passon.textFiles.length; i++) {
+            let element = passon.textFiles[i];
+            const fileBuffer = fs.readFileSync(element);
+            const charsetMatch = detectCharacterEncoding(fileBuffer);
+            debug('File %s has encoding %s with confidence %s', element, charsetMatch.encoding, charsetMatch.confidence);
+
+            // If encoding type is not supported (e.g. ISO)
+            if (config.encoding.supportedEncodings.indexOf(charsetMatch.encoding) === -1) {
+                debug('Unsupported encoding %s found in file %s!', charsetMatch.encoding, element);
+                let err = new Error('encoding not supported');
+                err.status = 422;
+                err.msg = 'encoding not supported';
```

Comment:
That has the file name array within the string, I suggest parseable JSON and a more informative error message that can be used in the UI. Also, the path must not expose any directory structure above the ERC root directory, see https://github.com/o2r-project/o2r-muncher/blob/master/lib/rewrite-tree.js

```json
{
   "error":"Files with unsupported encoding detected. Only UTF-8 is supported.",
   "files":[
      {
         "path":"ejpmi/data/test.txt",
         "encoding":"Shift_JIS"
      }
   ]
}
```